apps/examples/hello/hello_main.c

```diff
@@ -71,43 +76,88 @@ int hello_main(int argc, char *argv[])
 #endif
 {
 	printf("Hello, World!!\n");
-	/* Call stay IOCTL to keep PM domain in current state because maybe we know we need to do some
-	 * non-frequent work with time gaps and dont want the board to sleep in between
-	 */
-	// pm_ioctl(PM_IOC_STAY);
+
+	int fd_lock, fd_unlock, fd_sleep;
```

Comment:
Since the application layer will be controlling with the help of pm_procfs.c, means that we don't need pm_ioctl.c anymore rite? If then, I will move out the logic and integrate with pm_procfs.c, also provide the rest which is necessary.

Reply:
After thinking, I think we still have to keep pm_ioctl.c, the power_xxx_write api will be the middle layer to reach pm_ioctl, since the application should not have any knowledge about PM state, and also for easier maintenance in the future. Please share your thoughts on this~

Reply:
1. Yes, the functionality of pm_ioctl.c has to be included in pmm_procfs.c
2. "ioctl" is not available as a syscall for procfs_operations (https://github.com/Samsung/TizenRT/blob/0f25fb2c02dc4964019bb736ca979af2e03306e9/os/include/tinyara/fs/procfs.h#L70 ), so we cannot use ioctl directly in our applications.
If you want that some procfs syscall gets transformed into a pm_ioctl call, we may be able to do that.
Could you clarify why we should do that AND/OR share some basic implementation of your idea please?

Reply:
Hi @amogh-samsung ,

2) Hmm, but my meaning here is to invoke pm_ioctl in the pm_procfs, and application will invoke syscalls as it is (which is present in the structure here you have provided).

Reply:
Okay, so you mean?
App -> pm_procfs.c -> pm_ioctl.c -> pm_stay/relax

If yes, why do you prefer this control flow?
The same requirement can be handled by write(fd, PM_NORMAL, 1)
Where fd is an open file descriptor for /proc/power/domains/0/power_lock OR power_unlock
Is there some advantage in ioctl that I am missing?

Reply:
> After thinking, I think we still have to keep pm_ioctl.c, the power_xxx_write api will be the middle layer to reach pm_ioctl, since the application should not have any knowledge about PM state, and also for easier maintenance in the future. Please share your thoughts on this~

My opinion is over here, if you think that it is redundant, I can also integrate to just work in pm_procfs.c~

Reply:
Ok, lets avoid the use of the intermediate ioctl.
As long as we can get the work done with pm_procfs.c